java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/RuntimeImageBuildStep.java

```diff
@@ -86,14 +87,12 @@ private String getBaseRuntimeImage(BuildContext buildContext, Artifact artifact)
 
     // Select runtime based on artifact type
 
-    if (artifact.getPath().normalize().equals(buildContext.getWorkspaceDir())
-        && artifact.getType() == APP_ENGINE_EXPLODED_WAR) {
-      // If the workspace directory itself is an app engine exploded war, use the compat runtime.
-      logger.info("Using base image '{}' for App Engine exploded WAR artifact", compatImageName);
+    if (artifact.getType() == APP_ENGINE_EXPLODED_WAR || artifact.getType() == EXPLODED_WAR) {
```

Comment:
I understand that XRT treats all exploded WARs as compat, but I don't think we need to continue that here, as long as our detection of `APP_ENGINE_EXPLODED_WAR` is sound.

Reply:
I'm worried that if we move away from that behavior it would break working deployment configs.  Are you sure that every compat deployment has a file at `WEB-INF/appengine-web.xml`? 

Another thing worth bearing in mind is that the jetty and tomcat runtimes haven't really been tested with exploded war deployments. This could mean that we'd be switching from deploying to a working compat runtime, to a non-working jetty/tomcat runtime.